### DISCUSSION TITLE
FAQ

### DISCUSSION THREAD

**Discussion by UltimaHoarder** [UltimaHoarder/UltimaScraper#584] [Q&A]
Make sure you're using Python 3.9.X otherwise I will ignore you.

**Comment by anthr76** [comment]


**Reply by UltimaHoarder** [reply]
I don't, sorry, someone else would have to make it.

**Reply by Arnoldwa** [reply]
When i scrape it says 'No posts found" but there are posts??? Anyway to fix?

**Reply by UltimaHoarder** [reply]
Which username are you trying to scrape?

**Reply by dhwish** [reply]
I am having a similar issue. Not all the content is showing up. Specifically the locked "paid" posts do not download or the locked messages. I'm probably doing something wrong... The FAQ said there was a way to get around the paywalls. Please help :) An example profile would be https://onlyfans.com/daissydoll10free

Also is there a way to scrape profiles without subscribing to them?

**Comment by Arnoldwa** [comment]
https://onlyfans.com/lildeecakes

**Comment by drickerby** [comment]
The script doesn't seem to want to scrape the images/videos I was sent in messages from an expired subscription. Am I doing something wrong? It's working for all currently subscribed users, but not the one I've let my subscription lapse on.

**Reply by fightcancer2** [reply]
Same on my old PC.  It won't even start on my new PC.

EDIT: the inability to scrape may have come from the fact that I entered a capital 'A' instead of a lowercase for "a = Everything."

**Comment by fightcancer2** [comment]
I installed Python 3.9 on my new PC.  When I run start_ofd.py, I get this error.

G:\OnlyFans-master>start_ofd.py
Traceback (most recent call last):
  File "G:\OnlyFans-master\start_ofd.py", line 2, in <module>
    import tests.main_test as main_test
  File "G:\OnlyFans-master\tests\main_test.py", line 1, in <module>
    from apis.onlyfans.onlyfans import auth_details
  File "G:\OnlyFans-master\apis\onlyfans\onlyfans.py", line 10, in <module>
    from dateutil.relativedelta import relativedelta
ModuleNotFoundError: No module named 'dateutil'

Not sure what I'm doing wrong.

This program doesn't work on my old PC any more either.  It immediately says completed after scraping each account.

EDIT: the inability to scrape on my old PC may be because I entered a capital 'A' instead of a lowercase for "a = Everything."

**Reply by UltimaHoarder** [reply]
Reinstall the requirements

**Reply by fightcancer2** [reply]
OK.  Here's the text I get after reinstalling.  Btw, how do I donate via crypto currency?

`G:\OnlyFans-master>pip install -r requirements.txt
Requirement already satisfied: requests in c:\users\fightcancer\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 1)) (2.25.1)
Requirement already satisfied: beautifulsoup4 in c:\users\fightcancer\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 2)) (4.9.3)
Requirement already satisfied: urllib3 in c:\users\fightcancer\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 3)) (1.26.2)
Requirement already satisfied: win32-setctime in c:\users\fightcancer\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 4)) (1.0.3)
Requirement already satisfied: pysocks in c:\users\fightcancer\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 5)) (1.7.1)
Collecting psutil
  Using cached psutil-5.7.3.tar.gz (465 kB)
Collecting python-dateutil
  Using cached python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
Collecting lxml
  Using cached lxml-4.6.2-cp39-cp39-win_amd64.whl (3.5 MB)
Collecting mergedeep
  Using cached mergedeep-1.3.0-py3-none-any.whl (6.3 kB)
Collecting jsonpickle
  Using cached jsonpickle-1.4.2-py2.py3-none-any.whl (36 kB)
Collecting deepdiff[murmur]
  Using cached deepdiff-5.0.2-py3-none-any.whl (60 kB)
Collecting numpy==1.19.3
  Using cached numpy-1.19.3-cp39-cp39-win_amd64.whl (13.3 MB)
Collecting ujson
  Using cached ujson-4.0.1-cp39-cp39-win_amd64.whl (44 kB)
Requirement already satisfied: chardet<5,>=3.0.2 in c:\users\fightcancer\appdata\local\programs\python\python39\lib\site-packages (from requests->-r requirements.txt (line 1)) (4.0.0)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\fightcancer\appdata\local\programs\python\python39\lib\site-packages (from requests->-r requirements.txt (line 1)) (2020.12.5)
Requirement already satisfied: idna<3,>=2.5 in c:\users\fightcancer\appdata\local\programs\python\python39\lib\site-packages (from requests->-r requirements.txt (line 1)) (2.10)
Requirement already satisfied: soupsieve>1.2; python_version >= "3.0" in c:\users\fightcancer\appdata\local\programs\python\python39\lib\site-packages (from beautifulsoup4->-r requirements.txt (line 2)) (2.1)
Collecting six>=1.5
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting ordered-set>=4.0.1
  Using cached ordered-set-4.0.2.tar.gz (10 kB)
Collecting mmh3; extra == "murmur"
  Using cached mmh3-2.5.1.tar.gz (9.8 kB)
Using legacy 'setup.py install' for psutil, since package 'wheel' is not installed.
Using legacy 'setup.py install' for ordered-set, since package 'wheel' is not installed.
Using legacy 'setup.py install' for mmh3, since package 'wheel' is not installed.
Installing collected packages: psutil, six, python-dateutil, lxml, mergedeep, jsonpickle, ordered-set, mmh3, deepdiff, numpy, ujson
    Running setup.py install for psutil ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\users\fightcancer\appdata\local\programs\python\python39\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\fightcancer\\AppData\\Local\\Temp\\pip-install-t4i3xb9u\\psutil\\setup.py'"'"'; __file__='"'"'C:\\Users\\fightcancer\\AppData\\Local\\Temp\\pip-install-t4i3xb9u\\psutil\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\fightcancer\AppData\Local\Temp\pip-record-3t0kcy5n\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\fightcancer\appdata\local\programs\python\python39\Include\psutil'
         cwd: C:\Users\fightcancer\AppData\Local\Temp\pip-install-t4i3xb9u\psutil\
    Complete output (39 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.9
    creating build\lib.win-amd64-3.9\psutil
    copying psutil\_common.py -> build\lib.win-amd64-3.9\psutil
    copying psutil\_compat.py -> build\lib.win-amd64-3.9\psutil
    copying psutil\_psaix.py -> build\lib.win-amd64-3.9\psutil
    copying psutil\_psbsd.py -> build\lib.win-amd64-3.9\psutil
    copying psutil\_pslinux.py -> build\lib.win-amd64-3.9\psutil
    copying psutil\_psosx.py -> build\lib.win-amd64-3.9\psutil
    copying psutil\_psposix.py -> build\lib.win-amd64-3.9\psutil
    copying psutil\_pssunos.py -> build\lib.win-amd64-3.9\psutil
    copying psutil\_pswindows.py -> build\lib.win-amd64-3.9\psutil
    copying psutil\__init__.py -> build\lib.win-amd64-3.9\psutil
    creating build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\runner.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_aix.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_bsd.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_connections.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_contracts.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_linux.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_memleaks.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_misc.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_osx.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_posix.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_process.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_sunos.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_system.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_testutils.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_unicode.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\test_windows.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\__init__.py -> build\lib.win-amd64-3.9\psutil\tests
    copying psutil\tests\__main__.py -> build\lib.win-amd64-3.9\psutil\tests
    running build_ext
    building 'psutil._psutil_windows' extension
    VisualStudio is not installed; get it from http://www.visualstudio.com/en-au/news/vs2015-preview-vs
    error: Microsoft Visual C++ 14.0 is required. Get it with "Build Tools for Visual Studio": https://visualstudio.microsoft.com/downloads/
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\users\fightcancer\appdata\local\programs\python\python39\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\fightcancer\\AppData\\Local\\Temp\\pip-install-t4i3xb9u\\psutil\\setup.py'"'"'; __file__='"'"'C:\\Users\\fightcancer\\AppData\\Local\\Temp\\pip-install-t4i3xb9u\\psutil\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\fightcancer\AppData\Local\Temp\pip-record-3t0kcy5n\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\fightcancer\appdata\local\programs\python\python39\Include\psutil' Check the logs for full command output.
WARNING: You are using pip version 20.2.3; however, version 20.3.3 is available.
You should consider upgrading via the 'c:\users\fightcancer\appdata\local\programs\python\python39\python.exe -m pip install --upgrade pip' command.

G:\OnlyFans-master>`

**Reply by UltimaHoarder** [reply]
You have to follow what it says in the text. You have to download buildtools.

**Reply by fightcancer2** [reply]
Thank you! I will do so and restart.  How do I donate please?

**Reply by UltimaHoarder** [reply]
.-. Which coin would you want to donate with?

**Reply by fightcancer2** [reply]
Bitcoin

**Comment by fightcancer2** [comment]
In the file README.md, it says, "Open and edit: `.profiles/default/auth.json`," but I believe path should be "...\.profiles\OnlyFans\default\auth.json."  Maybe I'm wrong.

**Reply by fightcancer2** [reply]
Also, I just got my BTC account sorted out, verified, etc.  What's the email or address where I can donate please?

**Reply by UltimaHoarder** [reply]
My btc address is
3A37gaYFjPXV8kNQX3DgMBhkRouvBozojc

Thank you :) and the path is correct.

**Reply by fightcancer2** [reply]
> My btc address is
> 3A37gaYFjPXV8kNQX3DgMBhkRouvBozojc

Did you get the donation?

**Reply by UltimaHoarder** [reply]
> > My btc address is
> > 3A37gaYFjPXV8kNQX3DgMBhkRouvBozojc
> 
> Did you get the donation?

Yes, I got it, thanks 👍

**Comment by avekifes** [comment]
I noticed that the newer versions take a lot longer to scrape and attempt to merge new metadata with old/legacy metadata every time. Is that normal?

**Reply by avekifes** [reply]
Also, it seems to update creation and modification times of files every time.

**Reply by UltimaHoarder** [reply]
Do you have overwrite = true on

**Reply by avekifes** [reply]
No, it's set to false.

**Reply by UltimaHoarder** [reply]
Reason why it's taking longer to scrape is because I've set the script to merge legacy metadata BUT, I haven't told the script to delete the legacy metadata after it's been merged yet.

I'll set to delete legacy metadata when I'm 100% sure metadata merging is fixed.

**Comment by ddavidle6** [comment]
is there a way to sort the images/videos based on the upload year? eg. folder containing images posted in 2020.

**Reply by UltimaHoarder** [reply]
Nope

**Reply by James-MF-Thomas** [reply]
Couldn’t you just add the {post-date} to the beginning of file name. Then they should go in order of time when select list by name in whatever OS you’re running

**Comment by lemonpillow** [comment]
when i try to scrape a user im subscribed to it has no content and just moves on to a different user when i didn't select all users

**Reply by UltimaHoarder** [reply]
What does the script look like

**Reply by lemonpillow** [reply]
![4e3e754590b5e2590646f9e0d628de84](https://user-images.githubusercontent.com/48212700/103142166-34b7d600-46ff-11eb-888e-6ebfe7839311.png)
like this

**Reply by lemonpillow** [reply]
and profile looks like this
![e5f502ad816433b299e5ba17a2d2f2b0](https://user-images.githubusercontent.com/48212700/103142174-67fa6500-46ff-11eb-9fad-e8f32663e4ef.png)

**Reply by UltimaHoarder** [reply]
Just let the script run.

**Reply by lemonpillow** [reply]
It stays like this stuck on downloading the 2 videos

**Reply by UltimaHoarder** [reply]
I don't know how to fix it unless I have access to the account

**Comment by stevasaurus** [comment]
Thanks, it works great on OF. I see Patreon listed as supported in the config, how do I set up an auth file to scrape from Patreon?

**Reply by UltimaHoarder** [reply]
Not supported
